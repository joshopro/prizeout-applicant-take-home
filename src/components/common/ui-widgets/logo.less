.logo {
	max-width: 100%;
	width: 7.5rem;
	display: block;
}
