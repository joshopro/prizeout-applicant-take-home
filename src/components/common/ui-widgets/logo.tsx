import React, { Fragment } from 'react';
import Classnames from 'classnames';

import './logo.less';

export const Logo: React.FC = (): React.ReactElement => {
    const classes: string = Classnames(`logo`),
        imgSrc = 'https://assets.prizeout.com/Prizeout-logo.svg';

    return (
        <Fragment>
            <img data-testid="logo" className={classes} src={imgSrc} alt="Prizeout logo" />
        </Fragment>
    );
};
