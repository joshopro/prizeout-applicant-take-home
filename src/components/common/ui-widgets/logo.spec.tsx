import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { Logo } from './logo';

describe('Test Logo Component', () => {
    const imgSrc = `https://assets.prizeout.com/Prizeout-logo.svg`;
    const logoAltText = `Prizeout logo`;
    const logoId = `logo`;

    // Matches snapshot
    test('Component matches snapshot', () => {
        const { asFragment } = render(<Logo />);

        expect(asFragment()).toMatchSnapshot();
    });

    // Logo has logo class
    test('Logo has image source', () => {
        render(<Logo />);

        expect(screen.getByTestId(logoId)).toHaveClass('logo');
    });

    // Logo has src attribute with string
    test('Logo has image source', () => {
        render(<Logo />);

        expect(screen.getByTestId(logoId)).toHaveAttribute('src', expect.stringContaining(imgSrc));
    });

    // Logo has alt text
    test('Logo has alt text', () => {
        render(<Logo />);

        expect(screen.getByTestId(logoId)).toHaveAccessibleName(logoAltText);
    });
});
