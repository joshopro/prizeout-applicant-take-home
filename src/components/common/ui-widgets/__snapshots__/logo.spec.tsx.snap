// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Logo Component Component matches snapshot 1`] = `
<DocumentFragment>
  <img
    alt="Prizeout logo"
    class="logo"
    data-testid="logo"
    src="https://assets.prizeout.com/Prizeout-logo.svg"
  />
</DocumentFragment>
`;
